### DISCUSSION TITLE
Trigger custom validation for one field only

### DISCUSSION THREAD

**Discussion by s-KaiNet** [react-hook-form/react-hook-form#4302] [Q&A] [answered]
I have `mode: "all"` for a form with a bunch of text fields with very basic `required` or `pattern` rules. However, I have one field, which uses custom async validation. I don't want to perform async validation on every text change for this specific field, instead, I would like to validate this field only when a user clicks "Submit".  In other words, I need a behavior, when all fields follow `mode: "all"` validation, but one uses `mode: "onSubmit"`

Is it possible to do it?

**Comment by bluebill1049** [comment, marked as answer]
you will have to build this component separately and use `trigger` to trigger the validation manually.

**Reply by s-KaiNet** [reply]
Thank you, Bill!   
Do you know, if a custom resolver with Yup will support my scenario or custom component is the only way to move forward?

**Reply by bluebill1049** [reply]
In v7 yes (not released), since we are providing what's been validated.

**Reply by s-KaiNet** [reply]
For potential googlers, I ended up with something like this [codesandbox](https://codesandbox.io/s/react-hook-form-v6-validationschema-forked-756qn).   

Basically, I attach a function to a context, which indicates, whether we should perform an async validation. I set a variable to true just before validation on submit,  then restore it to false.
